tools/system_libs.py

```diff
@@ -1861,13 +1867,7 @@ def add_library(libname):
   # C libraries that override libc must come before it
   if settings.PRINTF_LONG_DOUBLE:
     add_library('libprintf_long_double')
-  # Becuase libc_optz overrides parts of libc, it is not compatible with `LINKABLE`
-  # (used in `MAIN_MODULE=1`) because with that setting we use `--whole-archive` to
-  # include all system libraries.
-  # However, because libc_optz is a size optimization it is not really important
-  # when used with `MAIN_MODULE=1` (which links in all system libraries, leading
-  # to overheads far bigger than any savings from libc_optz)
-  if settings.SHRINK_LEVEL >= 2 and not settings.LINKABLE:
+  if settings.SHRINK_LEVEL >= 2:
```

Comment:
I thought you would need to replicate the same logic here as in `can_use()`...  but maybe it never gets to this line in some cases.

Reply:
Yeah, I was wrong... We do need the duplication. Added .